common/conf/conf.cpp

```diff
@@ -32,6 +57,33 @@ string GetEnv(const string &var_name, const string &default_value) {
 	}
 }
 
+MenderConfig config;
+
+error::Error MenderConfig::ProcessCmdlineArgs(const vector<string> &args) {
+	string config_path = paths::DefaultConfFile;
+	string fallback_config_path = paths::DefaultFallbackConfFile;
+
+	for (auto arg : args) {
+		if (arg.find("--config=") == 0) {
+			config_path = arg.substr(9, arg.size());
+		}
+		if (arg.find("--fallback-config=") == 0) {
+			fallback_config_path = arg.substr(18, arg.size());
+		}
+	}
+
+	auto ret = this->LoadFile(config_path);
+	if (!ret) {
+		// XXX: this->Reset() ???
+		ret = this->LoadFile(fallback_config_path);
```

Comment:
One clarification: The fallback is always loaded, but the settings in the main config override it per key.

Reply:
Yeah, right. It's a fallback, but it's read first. Thanks for the heads-up!